### PR TITLE
Exclude internal content from admin checklist and usage stats

### LINKED ISSUE #41335: Admin checklist indicates more progress than actual due to audit items
### Describe the bug

The admin checklist of a completely fresh enterprise instance makes me think i have set the instance up. 
image

### To Reproduce

`MB_JETTY_PORT=3007 java -jar 1.49.5.jar`

Run a recent ee version and the checklist is far too filled out.

```clojure
setup=> (state-for-checklist)
{:db-type :h2,
 :hosted? false,
 :configured {:email false, :slack false},
 :counts {:user 2, :card 75, :table 21}, ;; <- audit content here
;; the following indicates that there are dashboards, non-sample dbs (audit db connection), collections, models. 
;; It's a rich instance but solely because of the audit info
 :exists {:non-sample-db true, :dashboard true, :pulse false, :hidden-table false, :collection true, :model true}}
```

### Expected behavior

The audit items should not factor into this checklist for new admins

### Logs

_No response_

### Information about your Metabase installation

```JSON
1.49.5.jar
```

### Severity

p2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by calherries**
1. Fixes https://github.com/metabase/metabase/issues/41335
2. Follows up on https://github.com/metabase/metabase/pull/41522, excluding more things from anonymous usage stats: users, databases, tables, and fields. This new behaviour was approved [here](https://metaboat.slack.com/archives/C0641E4PB9B/p1713811895337679) on slack.

### Implementation details

I've created a new multimethod [`metabase.models.interface/exclude-internal-content-hsql`](https://github.com/metabase/metabase/blob/0b6ae3534cd94456699aed05ec0ebf4a4030576c/src/metabase/models/interface.clj#L724-L729) that is used in both `metabase.api.setup/state-for-checklist` and `metabase.analytics.stats/anonymous-usage-stats` to filter out internal content from their statistics and counts.

### CODE REVIEW

File: src/metabase/models/collection.clj [hunk 1]
```diff
@@ -944,6 +944,13 @@
           :read  (perms/collection-read-path collection-or-id)
           :write (perms/collection-readwrite-path collection-or-id))})))
 
+(defmethod mi/exclude-internal-content-hsql :model/Collection
+  [_model & {:keys [table-alias]}]
+  (let [maybe-alias #(h2x/identifier :field (some-> table-alias name) %)]
+    [:and
+     [:not= (maybe-alias :type) [:inline "instance-analytics"]]
```
Comment (qnkhuat) [inline src/metabase/models/collection.clj:951]: can we make `instance-analytics` an constant?

File: src/metabase/models/card.clj [hunk 2]
```diff
@@ -570,6 +571,10 @@
   [_card]
   [:name (serdes/hydrated-hash :collection) :created_at])
 
+(defmethod mi/exclude-internal-content-hsql :model/Card
+  [_model & {:keys [table-alias]}]
+  [:not= (h2x/identifier :field (some-> table-alias name) :creator_id) config/internal-mb-user-id])
```
Comment (qnkhuat) [inline src/metabase/models/card.clj:576]:
```suggestion
  [:not= (h2x/identifier :field table-alias :creator_id) config/internal-mb-user-id])
```
`h2x/identifier` will turn it to if it's non-nil so you don't need to do it.

Comment (qnkhuat) [reply src/metabase/models/card.clj:576]: same with all other places

**Review by qnkhuat — APPROVED** [review]
Makes sense to me. 

Just want to throw the idea out there that we can achieve similar result by defining a derived model like `:model/PublicCard` and use `t2/define-before-select` to automatically inject the filter clause. But that's maybe a bit overkill for this.

File: src/metabase/models/card.clj (hunk 2 above)
Comment (calherries) [reply src/metabase/models/card.clj:576]: The point of this is to call `name` so the function can take a keyword `table-alias`, which feels more consistent with HoneySQL than requiring `table-alias` to be a string.

Comment (qnkhuat) [reply src/metabase/models/card.clj:576]: `h2x/identifier` allows components to be both string or keyword; it'll convert everything to string in the end

Comment (calherries) [reply src/metabase/models/card.clj:576]: I just tested this, you're right. I could have sworn there was a test failing on the keyword somewhere though. I will try the test suite again

**Pull request merged**